### PR TITLE
[master] python3, tests update (shellCheck, pylintPy3k), moved tests to github Actions

### PR DESCRIPTION
**Pull request by fstagni**
Replaces https://github.com/DIRACGrid/Pilot/pull/97

### CODE REVIEW

**Comment by fstagni** [comment]
@wkrzemien @Andrew-McNab-UK @ptakha please review this one. Beware: for `master` branch.

**Comment by fstagni** [comment]
I need a review here. 
@wkrzemien @Andrew-McNab-UK @ptakha @aldbr @atsareg @chaen @chrisburr

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+dependencies:
```
Comment (chrisburr) [inline environment.yml]: Many of these only exist in conda-forge so you should specify the channels:
```suggestion
channels:
  - conda-forge
  - defaults
dependencies:
```

**Comment by fstagni** [comment]
Please, do re-review. 
@wkrzemien @Andrew-McNab-UK @ptakha @aldbr @atsareg @chaen @chrisburr 

Sorry for the large number of commits and changes. These changes have been tested in https://github.com/DIRACGrid/DIRAC/pull/4392

File: tests/CI/ci_test.sh [hunk 2]
```diff
@@ -1,34 +1,44 @@
...
-git checkout $PILOTBRANCH
-echo `pwd`
-cd ../..
-echo `pwd`
-
-echo -e '***' $(date -u) "**** Got the tests ****\n"
-
-source TestCode/Pilot/tests/CI/pilot_ci.sh 
+if ! cd Pilot; then
+  exit 1
```
Comment (chaen) [inline tests/CI/ci_test.sh]: Instead of testing each line you could use `set -e` for all that preparation section

File: tests/CI/pilot_ci.sh [hunk 3]
```diff
@@ -74,21 +74,20 @@ function PilotInstall(){
   default
 
   cwd=$PWD
-  cd $PILOTINSTALLDIR
-  if [ $? -ne 0 ]
-  then
-    echo -e 'ERROR: cannot change to ' $PILOTINSTALLDIR
-    return
+  if ! cd "$PILOTINSTALLDIR"; then
```
Comment (chaen) [inline tests/CI/pilot_ci.sh]: same as above `set -e`

Comment (fstagni) [reply tests/CI/pilot_ci.sh]: I know, but `set -e` might be set by the caller of these files. There might be a reason for which you prefer not having it hardcoded in here.

**Comment by fstagni** [comment]
I am going to merge this PR on Monday. Please, do re-review.

File: Pilot/dirac-install.py [hunk 4]
```diff
@@ -743,26 +761,28 @@ def getTarsLocation(self, project, module=None):
...
 
-  def getDiracOsLocation(self, project=None, diracosDefault=False):
+  def getDiracOsLocation(self, useVanillaDiracOS=False):
     """
     Returns the location of the DIRAC os binary for a given project for example: LHCb or DIRAC, etc...
-
-    :param str project: the name of the project
-    :param bool diracosDefault: flag to take diracos distribution from the default location
-
+    :param bool useVanillaDiracOS: flag to take diracos distribution from the default location
```
Comment (andresailer) [inline Pilot/dirac-install.py]:
```suggestion

    :param bool useVanillaDiracOS: flag to take diracos distribution from the default location
```

File: tests/CI/ci_test.sh [hunk 5]
```diff
@@ -1,34 +1,44 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 PILOTBRANCH=pilot_loggers
-DEBUG=True
 
 echo -e "**** Starting Pilot tests with PilotLogger ****\n"
 
-echo -e '***' $(date -u) "**** Getting the tests ****\n"
+echo -e "***' $(date -u) **** Getting the tests ****\n"
```
Comment (andresailer) [inline tests/CI/ci_test.sh]:
```suggestion
echo -e "*** $(date -u) **** Getting the tests ****\n"
```

File: tests/CI/ci_test.sh [hunk 6]
```diff
@@ -1,34 +1,44 @@
...
 
-mkdir -p $PWD/TestCode
-cd $PWD/TestCode
+mkdir -p "$PWD/TestCode"
+if ! cd "$PWD/TestCode"; then
+  exit 1
+fi
 
 git clone https://github.com/wkrzemien/Pilot.git
```
Comment (andresailer) [inline tests/CI/ci_test.sh]: Should this be DIRACGrid?

**Pull request merged**